packages/core/src/bus/Event.ts

```diff
@@ -0,0 +1,13 @@
+/* eslint-disable @typescript-eslint/explicit-member-accessibility */
+import { EventDispatcher } from './EventDispatcher';
+import { BusInstruction } from './BusInstruction';
+
+export abstract class Event<T> extends BusInstruction<T> {
+  public publish(dispatcher: EventDispatcher): Promise<void> {
+    return dispatcher.publish<T>(this);
+  }
+}
+
+export type EventType<T, R extends Event<T> = Event<T>> = new (
+  ...args: any[]
+) => R;
```

Comment:
Let's use `Constructor` as a postfix for such types

```suggestion
export type EventConstructor<T, R extends Event<T> = Event<T>> = new (
  ...args: unknown[]
) => R;
```